### DISCUSSION TITLE
Implementing auth0

### DISCUSSION THREAD

**Discussion by sohail-shrestha** [payloadcms/payload#2887] [Feature Requests & Ideas]
How do I implement auth0 in `PayloadCMS`?

I have tried [this](https://github.com/finkinfridom/payload-auth0-plugin) plugin. It is not plug-and-play.

Expected behavior:
1. Login with Auth0's universal login.
2. Auth0 is going to provide id_token and access_token to a callback url, from where I'd like to create a user if it doesn't exist.
3. access_token from auth0 should be used for authentication.

This is not possible when I set `auth.disableLocalStrategy` to false User's collection config.

**Comment by finkinfridom** [comment]
hey @sohail-shrestha, I'm the owner of the `auth0` plugin :)
As per documentation (https://github.com/finkinfridom/payload-auth0-plugin/), you should specify the `strategy` option in our `auth`-based collection.
Working with strategies let you manage scenarios (user lookup, user creation, user merge) when integrating with external IdP.
